tests/utils/image-manager/image-manager.spec.ts

```diff
@@ -0,0 +1,547 @@
+import * as stream from 'stream';
+import { AssertionError, expect } from 'chai';
+import { stub } from 'sinon';
+import * as tmp from 'tmp';
+import { delay } from '../../utils';
+import * as fs from 'fs';
+import * as fsAsync from 'fs/promises';
+import * as stringToStream from 'string-to-stream';
+import { balena, registerMocks, cleanupMocks } from './test-setup';
+import {
+	getFileCreatedDate,
+	getImage,
+	getImagePath,
+	getImageWritableStream,
+	getStream,
+	isImageFresh,
+} from '../../../src/utils/image-manager';
+import { resolve, extname } from 'path';
+import * as mockFs from 'mock-fs';
+import * as rimraf from 'rimraf';
+import { promisify } from 'util';
+import * as os from 'os';
+
+const clean = async () => {
+	await promisify(rimraf)(await balena.settings.get('cacheDirectory'));
+};
+
+// it's not 1 big describe, it's many describes
+// We will need to wrap all of them in a describe and add the cleanup in an afterAll I guess
+describe('image-manager', function () {
+	before(function () {
+		registerMocks();
+	});
+	after(function () {
+		cleanupMocks();
+	});
+
+	describe('.getStream()', () => {
+		describe('given the existing image', function () {
+			beforeEach(function () {
+				this.image = tmp.fileSync();
+				fs.writeSync(this.image.fd, 'Cache image', 0, 'utf8');
+
+				this.cacheGetImagePathStub = stub(getImagePath);
```

Comment:
This used to be `m.sinon.stub(cache, 'getImagePath')` with `cache = require('../build/cache')`, so I guess you need to do:
```suggestion
				this.cacheGetImagePathStub = stub(imageManager, 'getImagePath');
```
where `imageManager`  is something like:
```
import * as imageManager from '../???/build??/utils/image-manager';
```

Reply:
Why wouldn't what I have now work? It looks like a valid overload of `stub` 🤔

Reply:
Ah wait I see, it came from the *build* folder before

Reply:
Changed it to
```
import {
	getFileCreatedDate,
	getImage,
	getImagePath,
	getImageWritableStream,
	getStream,
	isImageFresh,
} from '../../../build/utils/image-manager';
```

Kept using the stub overload I have for now though. If there's a specific reason one is better than the other, please let me know

Reply:
I don't think that the tests will pass in the way you call the stub.
Afaik, the first arg needs to be an object, and the second arg needs to be the function that will get stubbed.
You can think of it like:
```
let originalFunc;
stub = (obj, methodName) => {
    originalFunc = obj[methodName];
    obj[methodName] = new StubbedFunc();
    return obj[methodName]
};
```

Reply:
Yup, you are right